src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CollectionBackedStoreProvider.cs

```diff
@@ -4,167 +4,74 @@
 
 using System;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
 using System.Diagnostics;
-using System.Security.Cryptography;
 using System.Security.Cryptography.X509Certificates;
-using Microsoft.Win32.SafeHandles;
+using System.Threading;
 
 namespace Internal.Cryptography.Pal
 {
     internal class CollectionBackedStoreProvider : IStorePal
     {
-        private readonly X509Certificate2[] _certs;
-        private static readonly SafeEvpPKeyHandle InvalidPKeyHandle = new SafeEvpPKeyHandle(IntPtr.Zero, false);
+        private readonly bool _cloneOnRead;
```

Comment:
Having a PAL method like IStorePal.CopyTo() clone or not clone based on some boolean passed at construction time feels like a design smell. IStorePal.CopyTo() semantics should be consistent across the board. Can't whoever needs the copying do the copying himself? Why does the Unix Pal need to do this and not the Windows Pal?